agent/consul/state/state_store.go

```diff
@@ -50,6 +50,24 @@ const (
 	watchLimit = 2048
 )
 
+// StoreConfig Handle a StoreConfig to handle particular behaviours
+type StoreConfig struct {
+	AllowNodeRenaming bool
+}
+
+// NewStoreConfig returns a new StoreConfig with values copied from other Config.
+// if config is not nil, will copy options from Config
```

Comment:
OPT: would be better to have lowercase "Config" as you refer to the variable

Reply:
DONE